fe/fe-core/src/main/java/com/starrocks/scheduler/TaskRunManager.java

```diff
@@ -57,16 +61,40 @@ public SubmitResult submitTaskRun(TaskRun taskRun, int priority) {
 
         String queryId = UUIDUtil.genUUID().toString();
         TaskRunStatus status = taskRun.initStatus(queryId, System.currentTimeMillis());
-        status.setPriority(priority);
+        status.setPriority(option.getPriority());
+        status.setMergeRedundant(option.isMergeRedundant());
         GlobalStateMgr.getCurrentState().getEditLog().logTaskRunCreateStatus(status);
-        long taskId = taskRun.getTaskId();
-
-        PriorityBlockingQueue<TaskRun> taskRuns = pendingTaskRunMap.computeIfAbsent(taskId,
-                u -> Queues.newPriorityBlockingQueue());
-        taskRuns.offer(taskRun);
+        arrangeTaskRun(taskRun, option.isMergeRedundant());
         return new SubmitResult(queryId, SubmitResult.SubmitStatus.SUBMITTED);
     }
 
+    // At present, only the manual and automatic tasks of the materialized view have different priorities.
+    // The manual priority is higher. For manual tasks, we do not merge operations.
+    // For automatic tasks, we will compare the definition, and if they are the same,
+    // we will perform the merge operation.
+    public void arrangeTaskRun(TaskRun taskRun, boolean mergeRedundant) {
+        if (!tryTaskRunLock()) {
+            return;
+        }
+        try {
+            long taskId = taskRun.getTaskId();
+            MinMaxPriorityQueue<TaskRun> taskRuns = pendingTaskRunMap.computeIfAbsent(taskId,
+                    u -> MinMaxPriorityQueue.orderedBy(TaskRun::compareTo).create());
+            if (mergeRedundant && taskRuns.size() >= 1) {
+                TaskRun lastTaskRun = taskRuns.peekLast();
+                if (lastTaskRun.getStatus().getDefinition().equals(taskRun.getStatus().getDefinition())) {
```

Comment:
1. The definition between task runs is different?
2. Why only support to merge with the last one? If the definition is the same, can it be merged with the task run in the middle?

Reply:
1. Yes, For materialized view scenarios, there may be some partition refreshes that lead to different definitions.
2. You are right. I use another way to implement it .